cli/src/pcluster/commands.py

```diff
@@ -196,96 +193,66 @@ def _validate_cluster_name(cluster_name):
         sys.exit(1)
 
 
-def create(args):  # noqa: C901 FIXME!!!
-    LOGGER.info("Beginning cluster creation for cluster: %s", args.cluster_name)
-    LOGGER.debug("Building cluster config based on args %s", str(args))
-
-    _validate_cluster_name(args.cluster_name)
-
-    # Build the config based on args
-    pcluster_config = PclusterConfig(
-        config_file=args.config_file, cluster_label=args.cluster_template, fail_on_file_absence=True
-    )
-    pcluster_config.validate()
-
-    # Automatic SIT -> HIT conversion, if needed
-    HitConverter(pcluster_config).convert()
+def _parse_config_file(config_file, fail_on_config_file_absence=True):
+    """
+    Parse the config file and initialize config_file and config_parser attributes.
 
-    # get CFN parameters, template url and tags from config
-    storage_data = pcluster_config.to_storage()
-    cfn_params = storage_data.cfn_params
+    :param config_file: The config file to parse
+    :param fail_on_config_file_absence: set to true to raise SystemExit if config file doesn't exist
+    """
+    if config_file:
+        default_config = False
+    elif "AWS_PCLUSTER_CONFIG_FILE" in os.environ:
+        config_file = os.environ["AWS_PCLUSTER_CONFIG_FILE"]
+        default_config = False
+    else:
+        config_file = utils.default_config_file_path()
+        default_config = True
+
+    if not os.path.isfile(config_file):
+        if fail_on_config_file_absence:
+            error_message = "Configuration file {0} not found.".format(config_file)
+            if default_config:
+                error_message += (
+                    "\nYou can execute the 'pcluster configure' command "
+                    "or see https://docs.aws.amazon.com/parallelcluster/latest/ug/configuration.html"
+                )
+            utils.error(error_message)
+        else:
+            LOGGER.debug("Specified configuration file %s doesn't exist.", config_file)
+    else:
+        LOGGER.debug("Parsing configuration file %s", config_file)
+    try:
+        return load_yaml_dict(file_path=config_file)
+    except Exception as e:
+        utils.error(
+            "Error parsing configuration file {0}.\nDouble check it's a valid Yaml file. "
+            "Error: {1}".format(config_file, str(e))
+        )
 
-    _check_for_updates(pcluster_config)
 
-    bucket_name = None
-    artifact_directory = None
-    cleanup_bucket = False
-    try:
-        cfn_client = boto3.client("cloudformation")
-        stack_name = utils.get_stack_name(args.cluster_name)
+def create(args):
+    """Create cluster."""
+    LOGGER.info("Beginning cluster creation for cluster: %s", args.cluster_name)
+    LOGGER.debug("CLI args: %s", str(args))
 
-        # merge tags from configuration, command-line and internal ones
-        tags = _evaluate_tags(pcluster_config, preferred_tags=args.tags)
+    if not args.disable_update_check:
+        utils.check_if_latest_version()
```

Comment:
Why `check_if_latest_version` is only in create_cluster command?

Reply:
We don't want to show a warning message "there is a new version of parallelcluster" for every command execution, but only when creating new clusters.